lua/rustaceanvim/lsp/targets.lua

```diff
@@ -0,0 +1,43 @@
+local T = {}
+
+local rustc_targets_cache = nil
+
+--- Get rustc targets, use cache if available
+---@return table<string, boolean>
+local function get_rustc_targets()
+  if rustc_targets_cache then
+    return rustc_targets_cache
+  end
+
+  local result = vim.system({ 'rustc', '--print', 'target-list' }):wait()
```

Comment:
suggestion: `vim.system():wait()` blocks Neovim's event loop.
It's generally better to use `vim.system` and pass in a callback, which is a `function(result)` where `result` is a `vim.SystemCompleted`.

This means that this function has to be changed, e.g. to `with_rustc_targets(callback: fun(targets))`.
It's a bit of a refactor, but I think it's doable.

Reply:
done